### PR TITLE
Added ability to favorite recipes and view favorites by meal type (w/ Muxuan)

### PR DESCRIPTION
**Pull request by omidafshar**
Added a link to favorite each recipe. The user is then directed to a page displaying their favorited recipes, categorized by meal type (Breakfast, lunch, and dinner)

### CODE REVIEW

File: db/migrate/20190425201933_add_columns_to_recipes.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AddColumnsToRecipes < ActiveRecord::Migration
```
Comment (nkompella) [inline db/migrate/20190425201933_add_columns_to_recipes.rb:1]: I like the way this was done! Its very clean

File: features/add_favorite.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Feature: Add favorite recipes
```
Comment (nkompella) [inline features/add_favorite.feature:1]: Comprehensive tests :)

File: .byebug_history [hunk 3]
```diff
@@ -0,0 +1,9 @@
+continue
+n
```
Comment (techprojsforfun) [inline .byebug_history:2]: for next iteration, let's remove this and DS_Store

File: app/models/favorited_recipe.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class FavoritedRecipe < Recipe
```
Comment (techprojsforfun) [inline app/models/favorited_recipe.rb:1]: nice use of Single Table Inheritance!

File: db/schema.rb [hunk 5]
```diff
@@ -11,17 +11,17 @@
...
 
-ActiveRecord::Schema.define(version: 20190423235125) do
+ActiveRecord::Schema.define(version: 20190426213534) do
 
   create_table "recipes", force: :cascade do |t|
-    t.datetime "created_at",   null: false
-    t.datetime "updated_at",   null: false
-    t.integer  "day"
-    t.integer  "slot"
+    t.datetime "created_at", null: false
```
Comment (techprojsforfun) [inline db/schema.rb:17]: will these be overwritten automatically when we apply the migrations? So we don't have to worry about merge conflicts

Comment (omidafshar) [reply db/schema.rb:17]: Yes, from my understanding, running rake db:migrate applies new migrations, which in turn rewrites the schema.rb file

**Comment by techprojsforfun** [comment]
Looks good, but why did you remove the coverage files? are you sure that we don't want the whole coverage/ directory in the final repo? 
Edit: We will re-add the coverage/ directory in different PR.

**Pull request merged**